lib/transproc/recursion.rb

```diff
@@ -30,13 +32,10 @@ module Recursion
     # @api public
     def array_recursion(value, fn)
       result = fn[value]
+      guarded = Transproc(:guard, -> v { v.is_a?(::Array) }, -> v { Transproc(:array_recursion, fn)[v] })
```

Comment:
this should be memoized otherwise every time you call it guard function gets created, same applies to hash_recursion